### PR TITLE
Update lz4

### PR DESCRIPTION
**Pull request by danlark1**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Build/Testing/Packaging Improvement

Short description (up to few sentences):
Update LZ4

### CODE REVIEW

File: dbms/src/Compression/CompressionCodecLZ4.cpp [hunk 1]
```diff
@@ -47,7 +48,9 @@ UInt32 CompressionCodecLZ4::doCompressData(const char * source, UInt32 source_si
 
 void CompressionCodecLZ4::doDecompressData(const char * source, UInt32 source_size, char * dest, UInt32 uncompressed_size) const
 {
-    LZ4::decompress(source, dest, source_size, uncompressed_size, lz4_stat);
+    if (LZ4_decompress_fast(source, dest, uncompressed_size) < 0)
```
Comment (danlark1) [inline dbms/src/Compression/CompressionCodecLZ4.cpp]: this is only for testing purporses

**Comment by alexey-milovidov** [comment]
```
../contrib/lz4/lib/lz4.c:102:9: warning: 'LZ4_DISABLE_DEPRECATE_WARNINGS' macro redefined [-Wmacro-redefined]
#define LZ4_DISABLE_DEPRECATE_WARNINGS /* due to LZ4_decompress_safe_withPrefix64k */
```

**Comment by danlark1** [comment]
> ```
> ../contrib/lz4/lib/lz4.c:102:9: warning: 'LZ4_DISABLE_DEPRECATE_WARNINGS' macro redefined [-Wmacro-redefined]
> #define LZ4_DISABLE_DEPRECATE_WARNINGS /* due to LZ4_decompress_safe_withPrefix64k */
> ```

With latest commit, this message should disappear
https://github.com/lz4/lz4/blob/7a4e3b1fac5cd9d4ec7c8d0091329ba107ec2131/lib/lz4.c#L105

**Comment by alexey-milovidov** [comment]
I have built on master with all updated submodules about now and got this message.

**Comment by danlark1** [comment]
Even with this commit?
https://github.com/yandex/ClickHouse/commit/3f1523a80a0a19c301e3d31fbb59948c9130c069

**Comment by alexey-milovidov** [comment]
You can see this warning in `ClickHouse build check` for the latest commit (click to the clang-8 build and download logs):
https://clickhouse-builds.s3.yandex.net/0/e914d0d67ae86fb2eb1536f670785ada2f58b836/build_log_416891604_1555767409.txt

**Comment by danlark1** [comment]
> You can see this warning in `ClickHouse build check` for the latest commit (click to the clang-8 build and download logs):
> https://clickhouse-builds.s3.yandex.net/0/e914d0d67ae86fb2eb1536f670785ada2f58b836/build_log_416891604_1555767409.txt

There is no more warning after this commit https://github.com/yandex/ClickHouse/commit/3f1523a80a0a19c301e3d31fbb59948c9130c069

https://clickhouse-builds.s3.yandex.net/0/3f1523a80a0a19c301e3d31fbb59948c9130c069/build_log_422758508_1556644422.txt

This commit was about an hour ago.

**Comment by alexey-milovidov** [comment]
Ok.

**Pull request merged**